### PR TITLE
Fix 'InvalidInstanceIDNotFound' exception not being handled

### PR DESCRIPTION
**Pull request by andersonvaf**
This fixes the issue described in https://github.com/voxpupuli/beaker-aws/issues/26.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Change Log
 
-[Unreleased](https://github.com/puppetlabs/beaker-aws/compare/0.8.1...master)
+## [0.9.1](https://github.com/puppetlabs/beaker-aws/tree/0.9.1) (2021-06-18)
```
Comment (bastelfreak) [inline CHANGELOG.md:3]: please don't update the changelog, it's autogenerated for each release based on closed issues and pull requests :)

File: lib/beaker-aws/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module BeakerAws
-  VERSION = '0.9.0'
+  VERSION = '0.9.1'
```
Comment (bastelfreak) [inline lib/beaker-aws/version.rb:2]: please don't update the version, we identify the next release version during the release process

**Comment by bastelfreak** [comment]
@trevor-vaughan are you able to review this?

File: CHANGELOG.md (hunk 1 above)
Comment (andersonvaf) [reply CHANGELOG.md]: oh, nice. changes dropped down :)

File: lib/beaker-aws/version.rb (hunk 2 above)
Comment (andersonvaf) [reply lib/beaker-aws/version.rb]: cool. reverted back.

**Comment by andersonvaf** [comment]
This is a log snippet after the changes proposed:

```
aws-sdk: Waiting for all hosts to be running
aws-sdk: Wait for node agent to be running
[Aws::EC2::Client 400 0.04942 0 retries] describe_instances(instance_ids:["i-050b72cdbd0db5354"]) Aws::EC2::Errors::InvalidInstanceIDNotFound The instance ID 'i-050b72cdbd0db5354' does not exist
aws-sdk: Error while trying to describe instance (id: i-050b72cdbd0db5354). Likely the instance is not created yet.
Instance agent not yet available
aws-sdk: Sleeping 2 seconds for attempt 1.
[Aws::EC2::Client 200 0.179779 0 retries] describe_instances(instance_ids:["i-050b72cdbd0db5354"])  
aws-sdk: Sleeping 4 seconds for attempt 2.
[Aws::EC2::Client 200 0.054373 0 retries] describe_instances(instance_ids:["i-050b72cdbd0db5354"])  
aws-sdk: Sleeping 8 seconds for attempt 3.
[Aws::EC2::Client 200 0.11613 0 retries] describe_instances(instance_ids:["i-050b72cdbd0db5354"])  
aws-sdk: Sleeping 16 seconds for attempt 4.
aws-sdk: Add tags for agent
```

**Comment by andersonvaf** [comment]
Hi :) Just a friendly update on this PR to check if I can get it merged and tagged :D

**Pull request merged**